eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -614,11 +621,26 @@ object Router {
                   stash: Stash,
                   rebroadcast: Rebroadcast,
                   awaiting: Map[ChannelAnnouncement, Seq[GossipOrigin]], // note: this is a seq because we want to preserve order: first actor is the one who we need to send a tcp-ack when validation is done
-                  privateChannels: Map[ShortChannelId, PrivateChannel],
+                  privateChannels: Map[ByteVector32, PrivateChannel], // indexed by channel id
+                  resolveScid: Map[ShortChannelId, ByteVector32], // scid to channel_id
```

Comment:
Shouldn't we make it more explicit that this map is only used for private channels (currently) and unconfirmed channels (in the future)? There's no reason to store the mapping for public, announced channels, it's already in `channels`, right?

Reply:
Done with 1c0f45355, also renamed the field. Added the comment, but it's week, I wish we could enforce that somehow. Well we could by making fields private, but then updating the case class would be horrible. 🤷‍♂️ 

> There's no reason to store the mapping for public, announced channels, it's already in channels, right?

Yes